File: design_patterns_study/230407_factory/go/abstractFactory/factory/abstract.go
package factory

// build a struct that has a method called "createFactory"
// this method returns the interface for the concrete factories

type AbstractFactoryPolygon struct{}

func (a AbstractFactoryPolygon) CreateFactory(color string) FactoryColorPolygon {
	if color == "blue" {
		return FactoryColorPolygonBlue{}
	} else if color == "red" {
		return FactoryColorPolygonRed{}
	} else if color == "green" {
		return FactoryColorPolygonGreen{}
	} else {
		return nil
	}
}

File: design_patterns_study/230407_factory/python/abstract_factory/abstract.py

from dataclasses import dataclass
import concrete

@dataclass
class AbstractFactoryPolygon:
    def create_factory(self, color: str) -> concrete.FactoryPolygonColor:
        if color == "blue":
            return concrete.FactoryPolygonBlue()
        elif color == "red":
            return concrete.FactoryPolygonRed()
        elif color == "green":
            return concrete.FactoryPolygonGreen()
        else:
            raise Exception("Sorry, we've got a problem...")
